Matchismo/Matchismo/Model/CardMatchingGame.mm

```diff
@@ -12,23 +12,45 @@
 
 @interface CardMatchingGame()
 @property (nonatomic, readwrite) NSInteger score;
-@property (nonatomic, readwrite) NSString * lastCardFlipDescription;
+@property (nonatomic, readwrite) NSAttributedString * lastMoveDescription;
 @property (nonatomic, strong) NSMutableArray<Card *> *cards;
 @property (nonatomic, strong) NSMutableArray<Card *> *chosenCards;
 @end
 
 @implementation CardMatchingGame
 
+constexpr int TWO_MATCH_BONUS = 4;
+constexpr int THREE_THREE_MATCH_BONUS = 5;
+constexpr int THREE_TWO_MATCH_BONUS = 3;
+constexpr int MISMATCH_PENALTY = 2;
+constexpr int COST_TO_CHOOSE = 1;
+
 - (NSMutableArray<Card *> *)cards {
-  if (!_cards) _cards = [[NSMutableArray<Card *> alloc] init];
+  if (!_cards) {
+    _cards = [[NSMutableArray<Card *> alloc] init];
+  }
   return _cards;
 }
 
 - (NSMutableArray<Card *> *)chosenCards {
-  if (!_chosenCards) _chosenCards = [[NSMutableArray<Card *> alloc] init];
+  if (!_chosenCards) {
+    _chosenCards = [[NSMutableArray<Card *> alloc] init];
+  }
   return _chosenCards;
 }
 
+- (void)setLastMoveDescription:(NSAttributedString *)lastMoveDescription {
+  _lastMoveDescription = lastMoveDescription;
+  if (![lastMoveDescription isEqualToAttributedString:[[NSAttributedString alloc] initWithString:@""]]) {
```

Comment:
This is slightly inefficient as every time we perform this comparison we're doing a heap allocation just to create an empty string to compare against.  Would be better to cache it somewhere - it could even be static and shard across all instances. 
Or, better, can you could just use `lastMoveDescription.length == 0`?